web/auth/rate_limiting.go

```diff
@@ -98,7 +98,14 @@ func NewRedisCounter(client redis.UniversalClient) Counter {
 }
 
 func (r *redisCounter) Increment(key string, timeLimit time.Duration) (int64, error) {
-	return r.Client.Incr(key).Result()
+	count, err := r.Client.Incr(key).Result()
+	if err != nil {
+		return 0, err
+	}
+	if count == 1 {
```

Comment:
Why do you count time from first failed password and not from last failed password ?